assets/src/hooks/useAllDetours.ts

```diff
@@ -0,0 +1,20 @@
+import { useEffect, useState } from "react"
+import { fetchDetours } from "../api"
+import { isOk } from "../util/result"
+import { SimpleDetourData } from "../models/detour"
+
+/**
+ * A hook to fetch all detours, organized by "Active", "Drafts", "Past"
+ * @returns a list of {@link SimpleDetours}, or null if loading.
+ */
+export const useAllDetours = (): SimpleDetourData | null => {
+  const [detours, setDetours] = useState<SimpleDetourData | null>(null)
+
+  useEffect(() => {
+    fetchDetours().then((detours) => {
+      isOk(detours) && setDetours(detours.ok)
+    })
+  }, [])
+
+  return detours
+}
```

Comment:
note: Oh I realize you should be able to implment this using `useApiCall` instead of by hand

Reply:
This changes the semantics a _little bit_, so you could keep that by processing the result of `useApiCall` like `usePatternsByIdForRoute` or `useNearestIntersectionFetchResult`

Reply:
Ok a couple attempts I've tried:

in `detourListPage`, I've replaced the reference to useAllDetours with:
```
  const { result: result } = useApiCall({
    apiCall: useCallback(async () => 
      fetchDetours(), [])
  })

  const detours = isOk(result) && result.ok
```
but result is not a permissible format for isOk, which expects a `Result`. 

Also tried just adjusting the hook, in a similar pattern to the other hooks you noted, but then no detours ever populated the screen.

```
export const useAllDetours = (): any =>
  useApiCall({
    apiCall: useCallback(async () => 
      fetchDetours(), [])
  })
```